gp2gp-translator/src/test/java/uk/nhs/adaptors/pss/translator/mapper/ImmunizationMapperTest.java

```diff
@@ -183,6 +205,90 @@ public void mapObservationToImmunizationWithUNKVaccineCode() {
         assertImmunizationWithDefaultVaccineCode(immunization);
     }
 
+    @Test
+    public void When_EhrCompositionWithNoPatConfidentialityCode_Expect_MetaFromConfidentialityServiceWithSecurity() {
+        final var metaWithSecurity = MetaFactory.getMetaFor(META_WITH_SECURITY, META_PROFILE);
+        when(
+            confidentialityService.createMetaAndAddSecurityIfConfidentialityCodesPresent(
+                eq("Immunization-1"),
+                confidentialityCodeCaptor.capture(),
+                confidentialityCodeCaptor.capture()
+            )
+        ).thenReturn(metaWithSecurity);
+        final var ehrExtract = unmarshallEhrExtract("immunization_with_ehrComposition_nopat_confidentiality_code.xml");
+
+        List<Immunization> immunizations = immunizationMapper.mapResources(
+            ehrExtract, getPatient(),
+            getEncounterList(), PRACTISE_CODE
+        );
+        var securityMeta = immunizations
+            .get(0)
+            .getMeta()
+            .getSecurity(NOPAT_URL_CODESYSTEM, NOPAT_CODE);
+
+        assertAll(
+            () -> {
+                assertThat(confidentialityCodeCaptor.getAllValues().get(0).isPresent())
+                    .isTrue();
+                assertThat(confidentialityCodeCaptor.getAllValues().get(0).get())
+                    .usingRecursiveComparison()
+                    .isEqualTo(getNoPatCV());
+            },
+            () -> {
+                assertThat(securityMeta).
+                    isNotNull();
+                assertThat(securityMeta.getDisplay())
+                    .isEqualTo(NOPAT_DISPLAY);
+            }
+        );
+    }
+
+    @Test
+    public void When_ObservationWithNoPatConfidentialityCode_Expect_MetaFromConfidentialityServiceWithSecurity() {
+        final var metaWithSecurity = MetaFactory.getMetaFor(META_WITH_SECURITY, META_PROFILE);
+        when(
+            confidentialityService.createMetaAndAddSecurityIfConfidentialityCodesPresent(
+                eq("Immunization-1"),
+                confidentialityCodeCaptor.capture(),
+                confidentialityCodeCaptor.capture()
+            )
+        ).thenReturn(metaWithSecurity);
+        final var ehrExtract = unmarshallEhrExtract("immunization_with_observation_nopat_confidentiality_code.xml");
+
+        List<Immunization> immunizations = immunizationMapper.mapResources(
+            ehrExtract, getPatient(),
+            getEncounterList(), PRACTISE_CODE
+        );
+        var securityMeta = immunizations
+            .get(0)
+            .getMeta()
+            .getSecurity(NOPAT_URL_CODESYSTEM, NOPAT_CODE);
+
+        assertAll(
+            () -> {
+                assertThat(confidentialityCodeCaptor.getAllValues().get(1).isPresent())
```

Comment:
I sometimes extract these values out into a separate variable above and leave a little comment next to `.get(1)` explaining what this is within the context of the test, for example `.get(1) // observationStatement.getConfidentialityCode()`. Just to aid with readability.

Reply:
oh missed this one, might be an idea, let me address that